distributed/worker_state_machine.py

```diff
@@ -353,8 +354,60 @@ class AddKeysMsg(SendMessageToScheduler):
 
 @dataclass
 class StateMachineEvent:
-    __slots__ = ("stimulus_id",)
+    __slots__ = ("stimulus_id", "handled")
     stimulus_id: str
+    #: timestamp of when the event was handled by the worker
+    # TODO switch to @dataclass(slots=True) and uncomment (requires Python >=3.10)
+    # handled: float | None = field(init=False, default=None)
```

Comment:
I assume this is the reason for the `__new__` method containing the `self.handled = None` assignment